### ISSUE TITLE
Do not expose internal symbols

### ISSUE THREAD

**Issue by bclement-ocp** [OCamlPro/alt-ergo#807]
We should go through and replace the use of "special" identifiers (e.g. `@mod`, `@/`, `@*`, etc.) with a different namespace (`Builtin of Hstring.t` works, or maybe we should just reuse `Op`).

```console
$ cat quoted2.smt2 
(set-logic ALL)
(declare-fun |@mod| (Int Int) Int)
(assert (exists ((x Int)) (distinct (mod 1 x) (|@mod| 1 x))))
(check-sat)
$ alt-ergo quoted2.smt2 --frontend dolmen
; File "quoted2.smt2", line 4, characters 1-12: Valid (0.4211) (2 steps) (goal g_1)

unsat
$ z3 quoted2.smt2
sat
```

Note that:

 - This is not a big issue with `@`-symbols because symbols starting with `@` are reserved for solver use in the SMT-LIB format
 - This *may* be an issue with the "fresh" `!kXXX" symbols that we generate that are *not* reserved for solver use. I initially thought we coud use `.kXXX` (and also `.?__` prefix instead of `!?__` prefix for skolem variables) but there are comments in the code staying "do not remove the underscores thay are important for the ordering" so… we'll have to carefully test this.

Actually having different namespaces for internal / user symbols (instead of relying on prefixes and the such) would be ideal, although that will also have an impact on order.

**Comment by Stevendeo** [comment]
Should we close this issue?

**Comment by bclement-ocp** [comment]
Is this handled correctly by Dolmen now? If so yes.

**Comment by Gbury** [comment]
This should have been fixed by Gbury/dolmen#193 (but that requires a non-yet released version of dolmen)

**Comment by bclement-ocp** [comment]
That's fine — we have #894 which will have to be taken care of before the next release.
